src/main/java/core/basesyntax/FigureSuplier.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSuplier {
+    public static final int FIGURE_COUNT = 4;
+    private Random random = new Random();
+    private ColorSupplier colors = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        String color = colors.getRandomColor();
+        switch (figureNumber) {
+            case 0:
+                int side = random.nextInt(15) + 1;
+                return new Square(color, side);
```

Comment:
To avoid using return in each case you can define variable on method level and return in only once in the end of this metod